packages/core/src/tickets.ts

```diff
@@ -0,0 +1,129 @@
+import type HoprCoreConnector from '@hoprnet/hopr-core-connector-interface'
+import type { Types } from '@hoprnet/hopr-core-connector-interface'
+import type Hopr from '.'
+import { UnacknowledgedTicket } from './messages/ticket/unacknowledged'
+
+type OperationSuccess = { status: 'SUCCESS'; receipt: string }
+type OperationFailure = { status: 'FAILURE'; message: string }
+type OperationError = { status: 'ERROR'; error: Error | string }
+export type OperationStatus = OperationSuccess | OperationFailure | OperationError
+
+class Tickets<Chain extends HoprCoreConnector> {
+  constructor(private node: Hopr<Chain>) {}
```

Comment:
I don't think this pattern is a good one.

We have a ton of 'namespace' classes that use a reference to the `node` as basically a global scope. This means that we can't easily mock out tests, we have a ton of indirection, and we can't make things private on the node as all of these namespace classes are using it.

If we must have a namespace for tickets, rather than having the API on the node, then we should limit the things that this namespace needs, so we can dependency inject, and stay somewhat loosely coupled.

Reply:
@peterbraden feel free to provide with examples on how to transform this file into something else, I don't have a strong opinion on this matter, just following the rest of the codebase style as I believe there is value to get from consistency and it makes it easier for the original authors

Reply:
I was trying to move all functions that are publicly callable into a flattened index.ts - this also makes clearer what we are exposing to our applications.

I also tried being more specific in the network code, with the dependencies.

Consistency shouldn't be an excuse for not trying to improve things - I think the namespace classes approach isn't working across our codebase, and we should attempt to move away from it.

Reply:
@peterbraden let's talk about this through DMs